src/screens/Home.tsx

```diff
@@ -1,39 +1,41 @@
-import { StatusBar } from "expo-status-bar";
-import React, { useState } from "react";
-import { StyleSheet, Text, View, Pressable } from "react-native";
 import { useNavigation } from "@react-navigation/native";
+import { StatusBar } from "expo-status-bar";
+import { useState } from "react";
+import { Pressable, StyleSheet, Text, View } from "react-native";
 import SearchBar from "../components/SearchBar";
-import { getLocationsAsync, useAppDispatch } from "@store";
+import SearchButton from "../components/SearchButton";
 
 type Nav = {
   navigate: (value: string) => void;
 };
 
 export default function Home() {
   const [searchText, setSearchText] = useState("");
-  const navigation = useNavigation<Nav>();
-  const dispatch = useAppDispatch();
 
-  const getLocation = () => {
-    dispatch(getLocationsAsync());
+  const navigation = useNavigation<Nav>();
 
-    navigation.navigate("MapList");
+  const onSearchSubmit = () => {
+    alert(searchText);
   };
 
   return (
     <View style={styles.container}>
       <Pressable onPress={() => navigation.navigate("Info")}>
         <Text style={styles.title}>HotSoup</Text>
       </Pressable>
-      <Pressable style={styles.btnContainer} onPress={getLocation}>
+      <Pressable
+        style={styles.btnContainer}
+        onPress={() => navigation.navigate("MapList")}
+      >
         <Text style={styles.btnText}>See Meals Nearby</Text>
       </Pressable>
       <Text style={styles.text}>or</Text>
+
       <SearchBar
         searchText={searchText}
         onSearchChange={(newSearchText) => setSearchText(newSearchText)}
-        onSearchSubmit={() => alert(searchText)}
       />
+      <SearchButton onSearchSubmit={onSearchSubmit} />
```

Comment:
The two search buttons are different, but I see you reused the same one on both the Home screen and the MapList screen - this is incorrect. Please fix this. Remove the incorrect search button from the Home page, and update the searchButton.tsx file name to be "mapListSearchButton.tsx" and only use it on the MapList page